handlers/new.go

```
@@ -44,17 +42,21 @@ func generateCluster() string {
 	return hex.EncodeToString(b)
 }
 
-func Setup(etcdHost, disc string) {
+func Setup(hostAddr string, etcdAddrs []string) {
+	etcdMachines = etcdAddrs
 	cfg = &client.Config{
-		Endpoints: []string{etcdHost},
+		Endpoints: etcdMachines,
 		Transport: client.DefaultTransport,
 		// set timeout per request to fail fast when the target endpoint is unavailable
 		HeaderTimeoutPerRequest: time.Second,
 	}
-
-	u, _ := url.Parse(etcdHost)
+	u, err := url.Parse(hostAddr)
+	if err != nil {
+		log.Println("u: ", u)
+		return
```

you don't need to return the error and fail the setup?